### PR TITLE
Guide to Discrete Mathematics Book Add

### PR DESCRIPTION
**Pull request by SenukDias**
## What does this PR do?
Add resource

## For resources

Guide to Discrete Mathematics.pdf

### Description

Guide to practice mathematics

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by SenukDias — COMMENTED** [review]
new

**Comment by Ashu104** [comment]
423:1-544:192  warning  Alphabetical ordering: swap l.544 and l.543  alphabetize-lists  remark-lint
  423:1-544:192  warning  Incorrect number of blank lines between last section and next heading  blank-lines-1-0-2  remark-lint
  550:1-55[6](https://github.com/EbookFoundation/free-programming-books/actions/runs/6369968862/job/17290659312?pr=9628#step:5:7):149  warning  Incorrect number of blank lines between last section and next heading  blank-lines-1-0-2  remark-lint

** for this error change alphabet order of your commit on line 192 and you too some blank lines, go emit them.

**Comment by eshellman** [comment]
Thanky you @Ashu104 for helping

**Review by Thenlie — CHANGES_REQUESTED** [review]
I believe this link is wrong and should not link to the free-programming-books GitHub repo. I think this should link to https://core.ac.uk/download/pdf/326762636.pdf instead or another external resource.

Also, please check the output of free-programming-books-lint and fix the errors. Update the title with `(changes made)` once the lint check is passing. Thanks!

**Review by SenukDias — COMMENTED** [review]
new

**Review by SenukDias — COMMENTED** [review]
new

**Comment by SenukDias** [comment]
now?!!

**Review by SenukDias — COMMENTED** [review]
link update

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -485,6 +485,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Foundations of Constructive Probability Theory](https://arxiv.org/pdf/1906.01803.pdf) - Yuen-Kwok Chan (PDF)
 * [Geometry with an Introduction to Cosmic Topology](https://mphitchman.com) - Michael P. Hitchman
 * [Graph Theory](http://compalg.inf.elte.hu/~tony/Oktatas/TDK/FINAL/)
+* [Guide to Discrete Mathematics Book](https://core.ac.uk/download/pdf/326762636.pdf) - David Gries, Fred B. Schneider (PDF)
```
Comment (Thenlie) [inline books/free-programming-books-subjects.md:488]:
```suggestion
* [Guide to Discrete Mathematics](https://core.ac.uk/download/pdf/326762636.pdf) - David Gries, Fred B. Schneider (PDF)
```

**Review by Thenlie — CHANGES_REQUESTED** [review]
Thank you for the contribution! It does not look like the word "book" is actually in the title. Once that is fixed I believe this is ready to merge.

Please update the PR title with `(changes made)` once you have made this change.

**Review by Thenlie — APPROVED** [review]

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**